realm-jni/src/io_realm_internal_SharedGroup.cpp

```diff
@@ -52,28 +77,11 @@ JNIEXPORT jlong JNICALL Java_io_realm_internal_SharedGroup_nativeCreate(
 #endif
         }
         else {
-            SharedGroup::DurabilityLevel level;
-            if (durability == 0)
-                level = SharedGroup::durability_Full;
-            else if (durability == 1)
-                level = SharedGroup::durability_MemOnly;
-            else if (durability == 2)
-#ifdef _WIN32
-                level = SharedGroup::durability_Full;   // For Windows, use Full instead of Async
-#else
-                level = SharedGroup::durability_Async;
-#endif
-            else {
-                ThrowException(env, UnsupportedOperation, "Unsupported durability.");
-                return 0;
-            }
+            // Exception thrown for wrong durability value
+            SharedGroup::DurabilityLevel level = jint_to_durability_level(env, durability);
```

Comment:
you should check for != null here and return, throwing a Java exception will not abort C++ code execution, so if this fails you will send an illegal durability to SharedGroup most likely crashing it hard.